### ISSUE TITLE
partialLoop/partial View Helper can not be nested when using setObjectKey

### ISSUE THREAD

**Issue by tylkomat** [zendframework/zf2#3758]
Since always the same `partial(Loop)` instance is used, setting a new object key inside a `partial(Loop)` overwrites also the current object key (because same instance). Maybe it is a better way to provide the `objectKey` as optional parameter for the `__invoke` methods of `partial` and `partialLoop`.

**Comment by larsnystrom** [comment]
Since I just encountered this problem I thought I'd chip in.

As a user it's easy to get around this problem by saving the current objectKey to a temporary variable and restoring when you're done:

	<?php
		$prevKey = $this->partialLoop()->getObjectKey();
		$this->partialLoop()->setObjectKey('newkey');
		echo $this->partialLoop('partial.phtml', $object);
		$this->partialLoop()->setObjectKey($prevKey);
	?>

This could be implemented by the framework using a stack of `objectKeys` in the `PartialLoop` helper, letting `setObjectKey()` push a key to the stack and `__invoke()` pop a key from the stack when the loop is completed.

That would of course cause a BC break for those relying on the `objectKey` staying set during multiple invocations of `PartialLoop`. 

A somewhat more complex idea is to use a nesting-level bound `objectKey`, meaning if `partialLoop()` is called from within another `partialLoop()` the inner `objectKey` would be different from the outer object key, but default to the outer `objectKey`. I have created a pull request which implements this.
